monkey/infection_monkey/island_api_client/http_island_api_client_factory.py

```diff
@@ -12,16 +13,21 @@
 
 class HTTPIslandAPIClientFactory(AbstractIslandAPIClientFactory):
     def __init__(
-        self, agent_event_serializer_registry: AgentEventSerializerRegistry, agent_id: AgentID
+        self,
+        agent_event_serializer_registry: AgentEventSerializerRegistry,
+        agent_id: AgentID,
+        lock: BasicLock,
```

Comment:
Consider narrowing this to `Lock`. Since `RLock` is a `BasicLock`, this would accept an `RLock`, but we don't really know what the implications of that would be.